### ISSUE TITLE
Incorrect Header size in farm-sdk

### ISSUE THREAD

**Issue by uttarayan21** [solana-labs/solana-program-library#2740]
The size of the Header struct mentioned at [farms/farm-sdk/src/refdb.rs#L40](https://github.com/solana-labs/solana-program-library/blob/master/farms/farm-sdk/src/refdb.rs#L40) is possibly incorrect.

The size of `ArrayString64` ( ArrayString<typenum::U64> ) is 64 + 1 bits as mentioned [here](https://docs.rs/arraystring/latest/arraystring/struct.ArrayString.html). 

Therefore the size [here](https://github.com/solana-labs/solana-program-library/blob/master/farms/farm-sdk/src/refdb.rs#L40) should be `74` instead of `73`.

In short
`size_of::<ArrayString64>()` is `65`
`ArrayString64::capacity()` is `64`

**Comment by askibin** [comment]
Hi Uttarayan, 

yes, ArrayString64 indeed occupies 65 bytes in memory - 64 are taken by utf8 string, and 1 is the size. But serialized length is only 64 bytes because data is stored as a fixed-size string that always occupies 64 bytes, so there is no need for a size byte. Why fixed-size string? Because RefDB records are all of the same sizes, this allows to efficiently add/delete/update individual records without a need to shift or expand data around as in the case with dynamic length strings, i.e. no need to decode/encode the entire data set, but just single record. Same story with metadata or state storing accounts - because data structures have fixed size strings, you can quickly read or update individual fields and save on compute budget.

I think this issue can be closed, but feel free to reach out if you have any other questions.

**Comment by uttarayan21** [comment]
Ah I see. The size byte isn't required since all strings are `ArrayString64`.
 Thanks for the detailed explanation.
